src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,88 @@
+package duke.command;
+
+import java.time.format.DateTimeParseException;
+
+import duke.Ui;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+
+/**
+ * The type Add command that adds a user-specified task to a given list of tasks.
+ */
+public class AddCommand extends Command {
+
+    /** Enum that specifies the type of task being added */
+    private final AddCommandType addType;
+    /** User inputted string */
+    private final String userInput;
+    /** List of tasks to run the command on */
+    private final TaskList tasks;
+
+    /**
+     * Instantiates a new Add command.
+     *
+     * @param addType   task type that is being added using AddCommand
+     * @param userInput user-inputted string
+     * @param tasks     list tasks to add the new task to
+     */
+    public AddCommand(AddCommandType addType, String userInput, TaskList tasks) {
+        this.addType = addType;
+        this.userInput = userInput;
+        this.tasks = tasks;
+    }
+
+    @Override
+    public void execute() {
+        Task newTask = null;
+        String[] inputs = this.userInput.split(" ", 2);
+        try {
+            switch (this.addType) {
+            case todo: {
+                newTask = new Todo(inputs[1]);
+                break;
+            }
+            case event: {
+                String[] args = inputs[1].split(" /at ", 2);
+                String[] datetimeArgs = args[1].split(" ", 2);
+                if (datetimeArgs.length == 1) {
+                    newTask = new Event(args[0], args[1]);
+                } else {
+                    newTask = new Event(args[0], datetimeArgs[0], datetimeArgs[1]);
+                }
+                break;
+            }
+            case deadline: {
+                String[] args = inputs[1].split(" /by ", 2);
+                String[] datetimeArgs = args[1].split(" ", 2);
+                if (datetimeArgs.length == 1) {
+                    newTask = new Deadline(args[0], args[1]);
+                } else {
+                    newTask = new Deadline(args[0], datetimeArgs[0], datetimeArgs[1]);
+                }
+                break;
+            }
```

Comment:
According to the coding standard, the format of a case block does not have curly braces, and only requires the colon. Your other switch block in `Storage.java` line 58 is correct, I would suggest following that as well.
<img width="1290" alt="Screenshot 2021-09-02 at 10 09 52 PM" src="https://user-images.githubusercontent.com/77873928/131858937-afd97184-e237-463c-b992-9b85c7775867.png">

Reply:
I initially included blocks because switch case does not allow same variables to be declared in different case, and putting a block fixes that. I'll try abstracting the variables out then 😅